core/rawdb/freezer_table.go

```diff
@@ -444,6 +448,7 @@ func (t *freezerTable) Append(item uint64, blob []byte) error {
 		offset:  newOffset,
 	}
 	// Write indexEntry
+	t.index.Seek(0, 2)
```

Comment:
This seek is not necessary because Write advances the file position. In general, it'd be nice to isolate the file opening code into its own function. We only need to seek to the end once after opening the file, which can be done in said function.

Reply:
Not sure how truncate behaves. Probably it's sane to do a seek after truncate too.

Reply:
I see, yes. Will add open + truncate methods that encapsulate that behavious, and corresponding tests.

Reply:
@fjl By the way, did you realise the two seeks there are on separate files? One index, one head?

Reply:
I have now encapsulated the file open and truncate methods so that writes always append and truncates (resizes) set the file pointer to the end. (the os Truncate method does not move file write pointer).